src/EnergyPlus/ScheduleManager.hh

```diff
@@ -328,6 +331,109 @@ namespace ScheduleManager {
 
     int GetNumberOfSchedules(EnergyPlusData &state);
 
+    struct CSVRow
+    {
+        std::string_view operator[](std::size_t index) const;
+        std::size_t rowEnd();
+        std::string delimiter{'*'}; // eg. comma in CSV
+        void readNextRow(std::istream &str);
+
+    private:
+        std::string m_line;
+        std::vector<int> m_data;
+    };
+
+    std::istream &operator>>(std::istream &str, CSVRow &data);
+
+    void PreProcessIDF(EnergyPlus::EnergyPlusData &state, int &SchNum, int NumCommaFileSchedules);
+
+    struct schedInputIdfPreprocessObject
+    {
+        std::string name;     // <- can inherit from parent sched class
+        std::string fileName; // <- can inherit from parent sched class
+        int ScheduleTypePtr{-1};
+        int columnOfInterest{-1};
+        int numHourlyValues{-1};
+        int MinutesPerItem{-1};
+        int rowsToSkip{-1};
+        bool FileIntervalInterpolated{false};
+        std::basic_string<char> columnSeperator{"*"};
+        int columnarDataIndex{-1}; // TODO : Figure out what to do here (size_t to int conversion)
+        int SchNum{-1};
+
+        schedInputIdfPreprocessObject(std::string name,     // <- can inherit from parent sched class
+                                      std::string fileName, // <- can inherit from parent sched class
+                                      int ScheduleTypePtr = -1,
+                                      int columnOfInterest = -1,
+                                      int numHourlyValues = -1,
+                                      int MinutesPerItem = -1,
+                                      int rowsToSkip = -1,
+                                      bool FileIntervalInterpolated = false,
+                                      std::basic_string<char> columnSeperator = "*",
+                                      int SchNum = -1)
+        {
+
+            this->name = name;
+            this->fileName = fileName;
+            this->ScheduleTypePtr = ScheduleTypePtr;
+            this->columnOfInterest = columnOfInterest - 1; // - 1 to get index to match with c++ containers (start at 0)
+            this->numHourlyValues = numHourlyValues;
+            this->MinutesPerItem = MinutesPerItem;
+            this->rowsToSkip = rowsToSkip;
+            this->FileIntervalInterpolated = FileIntervalInterpolated;
+            this->columnSeperator = columnSeperator;
+            this->SchNum = SchNum;
+        }
+        void setColumnarDataIndex(int incomingData)
+        {
+            this->columnarDataIndex = incomingData;
+        }
+    };
+
+    void PopulateSetOfFilenames(const std::vector<schedInputIdfPreprocessObject> &allIdfSchedData, std::set<std::string> &setOfFilenames);
+
+    struct PreProcessedColumn
```

Comment:
Objects of this struct will have a <Real64> vector (vals) that holds the actual schedule values from the CSV file.